dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/harvest/server/OAIRecordServiceBeanTest.java

```diff
@@ -0,0 +1,165 @@
+package edu.harvard.iq.dataverse.harvest.server;
+
+import com.google.common.collect.Lists;
+import edu.harvard.iq.dataverse.persistence.GlobalId;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.harvest.OAIRecord;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.persistence.EntityManager;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+@ExtendWith(MockitoExtension.class)
+class OAIRecordServiceBeanTest {
+
+    @InjectMocks
+    private OAIRecordServiceBean oaiRecordServiceBean;
+
+    @Mock
+    private EntityManager entityManager;
+
+    @Test
+    public void updateOaiRecordForDataset_ForUpdatedGuestbook() {
+        //given
+        Clock testClock = Clock.fixed(Instant.ofEpochMilli(12345678), ZoneId.of("UTC"));
```

Comment:
It would be better to move this and other numbers to constants, so that we could know in which place we should expect updated value without comparing the numbers.